dias/outfile_tracking.py

```diff
@@ -0,0 +1,183 @@
+"""Track output files and processed times."""
+
+import copy
+from ch_util import ephemeris
+import os
+import sqlite3
+
+from dias.exception import DiasException
+
+
+class FileTracker:
+    """Keep an index of output files and start- and stop- times associated to them."""
+
+    # name of the database files
+    _DB_FILE = "data_index.db"
+    _ARCHIVE_DB_FILE = "archive_index.db"
+
+    _CREATE_ARCHIVE_DB_TABLE = """CREATE TABLE IF NOT EXISTS files(
+                                start TIMESTAMP,
+                                stop TIMESTAMP,
+                                ntime INTEGER,
+                                time_step REAL,
+                                filename TEXT UNIQUE ON CONFLICT REPLACE)"""
```

Comment:
So the idea is that we factor out the file tracking and database engagement from all of the analyzers? So they all use filetracker?

Reply:
yey